include/deal.II/base/memory_space.h

```diff
@@ -29,15 +33,23 @@ namespace MemorySpace
    * Structure describing Host memory space.
    */
   struct Host
-  {};
-
+  {
+    using kokkos_space = ::Kokkos::HostSpace;
+  };
 
+  /**
+   * Structure describing the default memory space.
+   */
+  struct Default
+  {
+    using kokkos_space = ::Kokkos::DefaultExecutionSpace::memory_space;
+  };
 
   /**
    * Structure describing CUDA memory space.
    */
-  struct CUDA
-  {};
+  // FIXME Only enable if CUDA is enabled in deal.II.
```

Comment:
Yes, at the moment we use `CUDA` explicitly in a lot of places that I would not touch otherwise. That's why I'd rather do it in a separate pull request.